src/layers/s2-geometry-layer/s2-geometry-layer.js

```diff
@@ -0,0 +1,231 @@
+
+      if (token) {
+        data.push({
+          // keep a reference to the original data index
+          index,
+          data: allData[index],
+          token
+        });
+      }
+    }
+    return data;
+  }
+
+  updateLayerMeta(allData, getS2Token) {
+    return {};
+  }
+
+  getDefaultLayerConfig(props = {}) {
+    return {
+      ...super.getDefaultLayerConfig(props),
+
+      // filled
+      filled: true
+    };
+  }
+
+  /* eslint-disable complexity */
+  formatLayerData(datasets, oldLayerData, opt = {}) {
+    const {
+      colorScale,
+      colorDomain,
+      colorField,
+      color,
+      sizeField,
+      sizeScale,
+      sizeDomain,
+      visConfig: {sizeRange, colorRange}
+    } = this.config;
+
+    const {gpuFilter} = datasets[this.config.dataId];
+    const getS2Token = this.getPositionAccessor();
+    const {data} = this.updateData(datasets, oldLayerData);
+
+    // color
+    const cScale =
+      colorField &&
+      this.getVisChannelScale(
+        colorScale,
+        colorDomain,
+        colorRange.colors.map(c => hexToRgb(c))
+      );
+
+    // height
+    const sScale =
+      sizeField && this.getVisChannelScale(sizeScale, sizeDomain, sizeRange, 0);
+
+    const getFillColor = cScale
+      ? d => this.getEncodedChannelValue(cScale, d.data, colorField)
+      : color;
+
+    const getElevation = sScale
+      ? d => this.getEncodedChannelValue(sScale, d.data, sizeField, 0)
+      : 0;
+
+    return {
+      data,
+      getS2Token,
+      getFillColor,
+      getElevation,
+      getFilterValue: gpuFilter.filterValueAccessor()
+    };
+  }
+  /* eslint-enable complexity */
+
+  renderLayer(opts) {
+    const {
+      data,
+      gpuFilter,
+      interactionConfig,
+      mapState
+    } = opts;
+
+    const eleZoomFactor = this.getElevationZoomFactor(mapState);
+    const {config} = this;
+    const {visConfig} = config;
+
+    const updateTriggers = {
+      getElevation: {
+        sizeField: config.sizeField,
+        sizeRange: visConfig.sizeRange
+      },
+      getFillColor: {
+        color: config.color,
+        colorField: config.colorField,
+        colorRange: config.visConfig.colorRange,
+        colorScale: config.colorScale
+      },
+      getFilterValue: gpuFilter.filterValueUpdateTriggers
+    };
+
+    const defaultLayerProps = this.getDefaultDeckLayerProps(opts);
+
+    return [
+      new S2Layer({
+        ...defaultLayerProps,
+        ...interactionConfig,
+        ...data,
+        getS2Token: d => d.token,
+
+        // color
+        opacity: visConfig.opacity,
+        filled: visConfig.filled,
+
+        // highlight
+        autoHighlight: true,
+        highlightColor: HIGHLIGH_COLOR_3D,
+
+        // elevation
+        extruded: visConfig.enable3d,
+        elevationScale: visConfig.elevationScale * eleZoomFactor,
+
+        // render
+        pickable: true,
+
+        updateTriggers,
+        elevationLowerPercentile: visConfig.elevationPercentile[0],
```

Comment:
`elevationLowerPercentile` and `elevationUpperPercentile` is not supported

Reply:
See above^

Reply:
removed elevationLowerPercentile and elevationUpperPercentile